src/Api/EmailValidationV4.php

```diff
@@ -0,0 +1,282 @@
+            $fileData = ['filePath' => $filePath];
+        } else {
+            $fileData = [
+                'fileContent' => $filePath,
+                'filename' => 'file',
+            ];
+        }
+
+        $postDataMultipart = [];
+        $postDataMultipart[] = $this->prepareFile('file', $fileData);
+
+        $response = $this->httpPostRaw(sprintf('/v4/address/validate/preview/%s', $previewId), $postDataMultipart);
+        $this->closeResources($postDataMultipart);
+
+        return $this->hydrateResponse($response, CreateBulkPreviewResponse::class);
+    }
+
+    /**
+     * @param string $previewId ID given when the list created
+     *
+     * @return mixed|ResponseInterface
+     *
+     * @throws Exception
+     */
+    public function getBulkPreview(string $previewId)
+    {
+        Assert::stringNotEmpty($previewId);
+
+        $response = $this->httpGet(sprintf('/v4/address/validate/preview/%s', $previewId));
+
+        return $this->hydrateResponse($response, GetBulkPreviewResponse::class);
+    }
+
+    /**
+     * @param string $previewId ID given when the list created
+     *
+     * @return bool
+     */
+    public function deleteBulkPreview(string $previewId)
+    {
+        Assert::stringNotEmpty($previewId);
+
+        $response = $this->httpDelete(sprintf('/v4/address/validate/preview/%s', $previewId));
+
+        return 204 === $response->getStatusCode();
+    }
+
+    /**
+     * @param string $previewId ID given when the list created
+     *
+     * @return mixed|ResponseInterface
+     *
+     * @throws Exception
+     */
+    public function promoteBulkPreview(string $previewId)
+    {
+        Assert::stringNotEmpty($previewId);
+
+        $response = $this->httpPut(sprintf('/v4/address/validate/preview/%s', $previewId));
+
+        return $this->hydrateResponse($response, PromoteBulkPreviewResponse::class);
+    }
+
+    /**
+     * @param string $fieldName Field Name
+     * @param array  $filePath  ['fileContent' => 'content'] or ['filePath' => '/foo/bar']
+     *
+     * @return array File Data
+     */
+    private function prepareFile(string $fieldName, array $filePath): array
+    {
+        $filename = isset($filePath['filename']) ? $filePath['filename'] : null;
+
+        if (isset($filePath['fileContent'])) {
+            // File from memory
+            $resource = fopen('php://temp', 'r+');
+            fwrite($resource, $filePath['fileContent']);
+            rewind($resource);
+        } elseif (isset($filePath['filePath'])) {
+            // File form path
+            $path = $filePath['filePath'];
+
+            // Remove leading @ symbol
+            if (0 === strpos($path, '@')) {
+                $path = substr($path, 1);
+            }
+
+            $resource = fopen($path, 'r');
+        } else {
+            throw new InvalidArgumentException('When using a file you need to specify parameter "fileContent" or "filePath"');
```

Comment:
I believe it's safe to be deleted.

Given this is an internal (private) method, just accessible by this API class, it's my understanding this `throw` becomes an unreachable statement, as the other two conditionals are checking for the two allowed options, and the user doesn't get a chance to set them dynamically when calling the method.

Reply:
@DavidGarciaCat removed